jina/clients/python/grpc.py

```diff
@@ -27,49 +29,46 @@ def __init__(self, args: 'argparse.Namespace'):
             os.unsetenv('http_proxy')
             os.unsetenv('https_proxy')
         self.logger = JinaLogger(self.__class__.__name__, **vars(args))
+        self.configure_event_loop()
         self.logger.debug('setting up grpc insecure channel...')
         # A gRPC channel provides a connection to a remote gRPC server.
-        self._channel = grpc.insecure_channel(
+        self._channel = grpc.aio.insecure_channel(
             f'{args.host}:{args.port_expose}',
             options={
                 'grpc.max_send_message_length': -1,
-                'grpc.max_receive_message_length': -1,
+                'grpc.max_receive_message_length': -1
             }.items(),
         )
-        self.logger.debug('waiting channel to be ready...')
-        try:
-            grpc.channel_ready_future(self._channel).result(
-                timeout=(args.timeout_ready / 1000) if args.timeout_ready > 0 else None)
-        except grpc.FutureTimeoutError:
-            self.logger.critical(f'can not connect to the server at {args.host}:{args.port_expose} after '
-                                 f'{args.timeout_ready} ms, please double check the ip and grpc port number'
-                                 f' of the server')
-            raise GRPCServerError(f'can not connect to the server at {args.host}:{args.port_expose}')
-
-            # create new stub
-        self.logger.debug('create new stub...')
+        asyncio.get_event_loop().run_until_complete(self._channel.channel_ready())
         self._stub = jina_pb2_grpc.JinaRPCStub(self._channel)
 
         # attache response handler
         self.logger.success(f'connected to the gateway at {self.args.host}:{self.args.port_expose}!')
         self.is_closed = False
 
-    def call(self, *args, **kwargs):
+    def configure_event_loop(self):
+        try:
+            self.loop = asyncio.get_running_loop()
```

Comment:
what is the meaning of this? we try to see if there is an eventloop and if not we go for `uvloop`? In which context may this happen?

Reply:
This is an attempt to make the client work with an already running loop (e.g.- Jupyter). We'd need an additional loop-patching to make this work on Jupyter though

Reply:
would be good to add a comment to avoid someone removing since I am sure it won't be covered by tests